src/lib/swapper/manager/utils.ts

```diff
@@ -19,33 +21,38 @@ import type { TradeQuote } from 'lib/swapper/api'
 export const getRatioFromQuote = ({
   quote,
   feeAsset,
-  buyAssetFiatRate,
-  sellAssetFiatRate,
-  feeAssetFiatRate,
+  cryptoMarketDataById,
 }: {
   quote: TradeQuote<ChainId>
   feeAsset: Asset
-  buyAssetFiatRate?: string
-  sellAssetFiatRate?: string
-  feeAssetFiatRate?: string
+  cryptoMarketDataById: Partial<Record<AssetId, Pick<MarketData, 'price'>>>
+  assetsById: Partial<Record<AssetId, Asset>>
 }): number | undefined => {
-  const totalSellAmountFiat = bnOrZero(
+  const buyAssetUsdRate = cryptoMarketDataById[quote.buyAsset.assetId]?.price
+  const sellAssetUsdRate = cryptoMarketDataById[quote.sellAsset.assetId]?.price
+  const feeAssetUsdRate = cryptoMarketDataById[feeAsset.assetId]?.price
```

Comment:
Do we want to early bail in case `cryptoMarketDataById[quote.buyAsset.assetId]` is undefined (and the same for the sell asset/fee asset)?

Again, not introduced by this PR, but the occurrences of `rate ?? '0'` below are concerning - we aren't effectively able to distinguish loading/errored/success states

Reply:
agreed - also are these not `price`s rather than rates?

Reply:
yeah this stems from our type system where `AssetId` is not `keyof typeof cryptoMarketDataById` so the compiler has no idea whether the value we get for a given asset ID is defined, but I'm not sure how we would tackle this statically without some mega LSP-crashing generated type that includes every asset ID we support

Reply:
I'm tempted not to touch the handling of this in this PR because even`sumProtocolFeesToDenom` is handling missing asset prices by coalescing to `0`. We'd likely need to add some hefty modaic handling with `Err(...)` on missing asset. But this would spread throughout the selectors all the way to the UI so seems very much out of scope for now